### PR TITLE
Implement a feature of renaming files

### PR DESCRIPTION
**Pull request by kuzmavictor**

### CODE REVIEW

**Comment by kuzmavictor** [comment]
@dkalinin PTAL.

File: app/view/component/file-item/index.js [hunk 1]
```diff
@@ -34,24 +36,37 @@ export class FileItem extends DraggableComponent {
...
     this._htmlLiElement.appendChild(this._htmlDivElement);
     this._root.appendChild(this._htmlLiElement);
-    this._addEventHandler();
-    this._fileNameSpan = new Span(this._htmlDivElement);
-    this._fileNameSpan.text = this._name;
+    this._render();
+  }
+
+  /**
+   * Renders file item all its own fields.
```
Comment (dkalinin) [inline app/view/component/file-item/index.js]: Renders fields of file model.

File: app/view/component/file-item/index.js [hunk 2]
```diff
@@ -150,8 +229,10 @@ export class FileItem extends DraggableComponent {
       this._htmlDivElement.addEventListener('dragleave',
           (event) => this.onDragLeave(event, this._htmlDivElement));
       this._htmlDivElement.addEventListener('drop',
-          (event) => this.onDropToFolder(event));
+          (event) => this.onDrop(event));
     }
+    this._fileName.spanHtmlElement.addEventListener('dblclick',
```
Comment (dkalinin) [inline app/view/component/file-item/index.js]: this._fileName.onDoubleClick();

File: app/view/component/file-item/index.js [hunk 3]
```diff
@@ -63,6 +78,62 @@ export class FileItem extends DraggableComponent {
...
+   * Handlers action 'focusout' of input component. In fact it is the body
+   * of callback function.
+   *
+   * @param {Function} event - an callback function for `focusout` event for input component
+   * @private
+   */
+  _onFocusOutHandler(event) {
+    event.preventDefault();
+    event.stopPropagation();
+    if (!this.isRenameStart) {
```
Comment (dkalinin) [inline app/view/component/file-item/index.js]: Please remove isRenameStart.

File: app/service/store/action/index.js [hunk 4]
```diff
@@ -134,4 +135,26 @@ export const actions = {
...
+    store.requestServiceProvider.fileListService.renameFile(file)
+        .then((response) => {
+          store.dispatch('fetchFiles', store.state.files);
+        })
+        .catch((error) => {
+          console.log(error);
+          store.commit('setRenamingFileError', error);
+        })
+        .finally(() => {
+          store.commit('setOperationStatus', '');
```
Comment (dkalinin) [inline app/service/store/action/index.js]: It clears status from fetching files.

**Pull request merged**